File: internal/indexer/extract/generic/testdata/ruby/sample.rb
require "json"
require_relative "lib/base"

module Store
  # A user repository.
  class UserRepo < BaseRepo
    include Enumerable
    LIMIT = 10

    attr_reader :name

    def initialize(name)
      super()
      @name = name
    end

    def find(id)
      Helper.check(id)
      u = User.new(id)
      helper(id)
    end

    def self.create
      new("x")
    end

    private

    def reset
      @items.clear
    end
  end
end

def top_level(x)
  x + 1
end

class Widget
  private def secret
    1
  end
end

File: internal/indexer/extract/generic/testdata/objc/sample.m
#import <Foundation/Foundation.h>
#import "Repo.h"

/// A user repository.
@interface UserRepo : BaseRepo <Repo, NSCopying>
@property (nonatomic, strong) NSString *name;
- (instancetype)initWithName:(NSString *)name;
+ (UserRepo *)shared;
@end

@implementation UserRepo

- (instancetype)initWithName:(NSString *)name {
    self = [super init];
    _name = name;
    [Helper check:name];
    return self;
}

+ (UserRepo *)shared {
    return [[UserRepo alloc] initWithName:@"x"];
}

@end

@protocol Repo <NSObject>
- (id)find:(NSString *)key;
@end

static int counter = 0;

void helper(int n) {
    NSLog(@"%d", n);
}
